frontend/public/components/routes/create-route.tsx

```diff
@@ -0,0 +1,447 @@
+/* eslint-disable no-undef */
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Helmet } from 'react-helmet';
+import { Link } from 'react-router-dom';
+
+import { ButtonBar, Dropdown, history, resourcePathFromModel, ResourceName } from '../utils';
+import { k8sCreate, k8sList, K8sResourceKind } from '../../module/k8s';
+import { formatNamespacedRouteForResource, getActiveNamespace } from '../../ui/ui-actions';
+import { ServiceModel, RouteModel } from '../../models';
+import { AsyncComponent } from '../utils/async';
+
+const getPortOptions = (service: K8sResourceKind) => {
+  if (!service) {
+    return {};
+  }
+
+  const ports = _.get(service, 'spec.ports', []);
+  const portOptions = ports.reduce((acc, { name, port, targetPort, protocol }) => {
+    acc[name] = <React.Fragment>{port} &rarr; {targetPort} ({protocol})</React.Fragment>;
+    return acc;
+  }, {});
+
+  return portOptions;
+};
+
+const DroppableFileInput = (props) => <AsyncComponent loader={() => import('../utils/file-input').then(c => c.DroppableFileInput)} {...props} />;
+
+export class CreateRoute extends React.Component<null, CreateRouteState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formData: {
+        name: '',
+        hostname: '',
+        path: '',
+        service: '',
+        targetPort: '',
+        terminationType: '',
+        insecureTrafficType: '',
+        certificateFile: '',
+        privateKeyFile: '',
+        caCertificateFile: '',
+        destinationCaCertificateFile: '',
+        error: '',
+      },
+      toggleData: {
+        secure: false,
+      },
+      loaded: false,
+      inProgress: false,
+      namespace: getActiveNamespace(),
+      services: [],
+      labels: {},
+      portOptions: {},
+    };
+  }
+
+  componentDidMount() {
+    k8sList(ServiceModel, {ns: this.state.namespace})
+      .then(services => this.setState({
+        services,
+        loaded: true,
+      }))
+      .catch((err) => {
+        if (err.response.status !== 403) {
```

Comment:
I don't see 403 as a special case here. You still won't be able to create the route, so we should show an error.